Implement oekg predicates

madbkr opened the pull request:
## Summary of the discussion

As discussed in #2013 we need some more predicates to use in the OEKG. Some of those are implemented here as discussed in the issue.

## Type of change (CHANGELOG.md)

### Add
has iri, has study descriptor tag, has uuid, has scenario type, has publication date, has report title, has doi, has reference, based on sector division, has creation date, covers technology

## Workflow checklist

### Automation
Closes #

### PR-Assignee
- [ x] 🐙 Follow the [Pull Request Workflow](https://github.com/OpenEnergyPlatform/ontology/wiki/Pull-request-workflow)
- [x ] 📝 Update the [CHANGELOG.md](https://github.com/OpenEnergyPlatform/ontology/blob/dev/CHANGELOG.md)
- [x ] 📙 Add #'s to `term tracker annotation`

### Reviewer
- [ ] 🐙 Follow the [Reviewer Guide](https://github.com/OpenEnergyPlatform/ontology/wiki/Pull-request-workflow#reviewer-guide-check-changes-introduced-by-a-pull-request)
- [ ] 🐙 Provided feedback and show sufficient appreciation for the work done

stap-m commented on src/ontology/edits/oeo-model.omn:
```diff
@@ -549,6 +561,134 @@ pull request: https://github.com/OpenEnergyPlatform/ontology/pull/1944",
+        <http://purl.obolibrary.org/obo/IAO_0000027> or <http://purl.obolibrary.org/obo/IAO_0000310>
+    
+    Range: 
+        OEO_00140178 some rdfs:Literal
```
```suggestion
        OEO_00140178 some rdfs:Literal
```
why has number? A literal can also be a string... https://www.w3.org/TR/rdf-schema/#ch_literal

I have the impression that has iri should be a data property itself. Like this, the range could be rdfs:literal directly.

stap-m commented on src/ontology/edits/oeo-model.omn:
```diff
@@ -549,6 +561,134 @@ pull request: https://github.com/OpenEnergyPlatform/ontology/pull/1944",
+        OEO_00020227
+    
+    Range: 
+        OEO_00140178 some rdfs:Literal
```
Maybe we need a new data property for that. Has literal value?

madbkr replied:
To be honest, I wasn't quite sure how to implement it as `has number` didn't quite fit. I think making another data property to fix it is a good idea.

stap-m replied:
Sure. has iri, has uuid, has publication date, has report title, has doi, has reference I guess.

stap-m reviewed (changes requested):
My remarks on has number and object property vs data property seem to apply to more than `has iri`.

stap-m commented on src/ontology/edits/oeo-model.omn:
```diff
@@ -621,6 +761,22 @@ pull request: https://github.com/OpenEnergyPlatform/ontology/pull/1834",
+        OEO_00020426 "add
+issue: https://github.com/OpenEnergyPlatform/ontology/issues/2013
+pull request: https://github.com/OpenEnergyPlatform/ontology/pull/2020",
+        <http://purl.obolibrary.org/obo/IAO_0000115> "Has literal value is a data property that links an entity to a type of literal that describes it.",
```
```suggestion
        <http://purl.obolibrary.org/obo/IAO_0000115> "Has literal value is a data property that links an entity to a value of type literal.",
```

stap-m replied:
With making the relevant predicates data properties, the has literal value is not needed at the moment.

stap-m commented:
Does it work again @madbkr ?

madbkr commented:
@stap-m 
Yes, thank you!

madbkr commented:
@stap-m 
A few notes about the changes:

I made most of the new properties data properties.

As discussed I added a separate one for the publication vs creation date. I tried doing it with a custom data type but our testing pipeline does not support that and the check will fail so I removed it again.

I added a new sub-property for covers technology, but looking at the domains/ranges that may be another case of one that needs a shortcut. The domain/ranges for covers do not match the ones we need. So maybe this relation is pointless as we will need the shortcut anyway? Let me know what you think.

stap-m commented on src/ontology/edits/oeo-model.omn:
```diff
@@ -549,6 +561,75 @@ pull request: https://github.com/OpenEnergyPlatform/ontology/pull/1944",
+        OEO_00020426 "add
+issue: https://github.com/OpenEnergyPlatform/ontology/issues/2013
+pull request: https://github.com/OpenEnergyPlatform/ontology/pull/2020",
+        <http://purl.obolibrary.org/obo/IAO_0000115> "has study descriptor tag is a relation between a scenario bundle and a oeo class represented by a study descriptor that is tagging it.
```
```suggestion
        <http://purl.obolibrary.org/obo/IAO_0000115> "has study descriptor tag is a relation between a scenario bundle and an oeo class which is used as a study descriptor.",
```

stap-m replied:
I'd propose to add an elucidation annotation with a text like that:
"OEO classes that are used as study descriptors are annotated by the annotation property "oekg annotation" with value "study descriptor".

stap-m commented on src/ontology/edits/oeo-model.omn:
```diff
@@ -549,6 +561,75 @@ pull request: https://github.com/OpenEnergyPlatform/ontology/pull/1944",
+pull request: https://github.com/OpenEnergyPlatform/ontology/pull/2020",
+        <http://purl.obolibrary.org/obo/IAO_0000115> "has study descriptor tag is a relation between a scenario bundle and a oeo class represented by a study descriptor that is tagging it.
+
+Only classes with the oekg annotation property are eligible.",
```
```suggestion
```

stap-m commented on src/ontology/edits/oeo-model.omn:
```diff
@@ -549,6 +561,75 @@ pull request: https://github.com/OpenEnergyPlatform/ontology/pull/1944",
+        OEO_00020426 "add
+issue: https://github.com/OpenEnergyPlatform/ontology/issues/2013
+pull request: https://github.com/OpenEnergyPlatform/ontology/pull/2020",
+        <http://purl.obolibrary.org/obo/IAO_0000115> "has scenario type is the relation between a scenario factsheet and the scenario-subclass it is about.",
```
```suggestion
        <http://purl.obolibrary.org/obo/IAO_0000115> "has scenario type is a relation between a scenario factsheet and a subclass of scenario.",
```

stap-m commented on src/ontology/edits/oeo-model.omn:
```diff
@@ -549,6 +561,75 @@ pull request: https://github.com/OpenEnergyPlatform/ontology/pull/1944",
+pull request: https://github.com/OpenEnergyPlatform/ontology/pull/2020",
+        <http://purl.obolibrary.org/obo/IAO_0000115> "has reference is the relation between a continuant and a reference that points to a relevant resource to the continuant.",
+        rdfs:label "has reference"@en
+    
```
I suggest to add an example of usage annotation with two or three examples (spatial region/country code, document/link to document, ...)

stap-m commented on src/ontology/edits/oeo-model.omn:
```diff
@@ -549,6 +561,75 @@ pull request: https://github.com/OpenEnergyPlatform/ontology/pull/1944",
+        OEO_00020426 "add
+issue: https://github.com/OpenEnergyPlatform/ontology/issues/2013
+pull request: https://github.com/OpenEnergyPlatform/ontology/pull/2020",
+        <http://purl.obolibrary.org/obo/IAO_0000115> "covers technology is a relation that holds between a study and the technology it covers.",
```
```suggestion
        <http://purl.obolibrary.org/obo/IAO_0000115> "covers technology is a relation that holds between a study and a technology covered in the study.",
```

stap-m commented on src/ontology/edits/oeo-model.omn:
```diff
@@ -621,6 +702,108 @@ pull request: https://github.com/OpenEnergyPlatform/ontology/pull/1834",
+        OEO_00020426 "add
+issue: https://github.com/OpenEnergyPlatform/ontology/issues/2013
+pull request: https://github.com/OpenEnergyPlatform/ontology/pull/2020",
+        <http://purl.obolibrary.org/obo/IAO_0000115> "has iri is a relation between a data item or document and the IRI that identifies it as rescource on the internet.",
```
should this relation be a subproperty of has reference? do we need both?

madbkr replied:
This is currently not possible as `has reference` is an object property and the other two are data properties.

stap-m commented on src/ontology/edits/oeo-model.omn:
```diff
@@ -621,6 +702,108 @@ pull request: https://github.com/OpenEnergyPlatform/ontology/pull/1834",
+        OEO_00020426 "add
+issue: https://github.com/OpenEnergyPlatform/ontology/issues/2013
+pull request: https://github.com/OpenEnergyPlatform/ontology/pull/2020",
+        <http://purl.obolibrary.org/obo/IAO_0000115> "has uuid is the relation between a factsheet or report and the 128-bit label used to identify it in a computer system.",
```
same here: subproperty of has reference?

stap-m commented on src/ontology/edits/oeo-model.omn:
```diff
@@ -621,6 +702,108 @@ pull request: https://github.com/OpenEnergyPlatform/ontology/pull/1834",
+        OEO_00020426 "add
+issue: https://github.com/OpenEnergyPlatform/ontology/issues/2013
+pull request: https://github.com/OpenEnergyPlatform/ontology/pull/2020",
+        <http://purl.obolibrary.org/obo/IAO_0000115> "has publication date is the relation between a report and the year it was published at.",
```
```suggestion
        <http://purl.obolibrary.org/obo/IAO_0000115> "has publication date is the relation between a report and the year it was published in.",
```

stap-m commented on src/ontology/edits/oeo-model.omn:
```diff
@@ -621,6 +702,108 @@ pull request: https://github.com/OpenEnergyPlatform/ontology/pull/1834",
+        OEO_00020426 "add
+issue: https://github.com/OpenEnergyPlatform/ontology/issues/2013
+pull request: https://github.com/OpenEnergyPlatform/ontology/pull/2020",
+        <http://purl.obolibrary.org/obo/IAO_0000115> "has doi is the relation between a report and the literal representing its doi.",
```
```suggestion
        <http://purl.obolibrary.org/obo/IAO_0000115> "has doi is the relation between a digital object and the literal representing its doi (digital object identifier).",
```

stap-m commented on src/ontology/edits/oeo-model.omn:
```diff
@@ -600,7 +603,13 @@ ObjectProperty: OEO_00390078
-        <http://purl.obolibrary.org/obo/IAO_0000115> "has reference is the relation between a continuant and a reference that points to a relevant resource to the continuant.",
+        <http://purl.obolibrary.org/obo/IAO_0000115> "has reference is the relation between a continuant and a reference that points to a relevant resource to the continuant.
+
+Example:
```
Please use the "example of usage" annotation property
